ubcpi/answer_pool.py

```diff
@@ -0,0 +1,315 @@
+
+    Args:
+        answers (str): the answers to be checked
+        options (dict): all options that should exist in the answers
+        algo (str): selection algorithm
+
+    Returns:
+        None if everything is good. Otherwise, the missing option error message.
+    """
+    seen_options = {}
+    for answer in answers:
+        if answer:
+            key = options[answer['answer']].get('text')
+            if options[answer['answer']].get('image_url'):
+                key += options[answer['answer']].get('image_url')
+            seen_options.setdefault(key, 0)
+            seen_options[key] += 1
+
+    missing_options = []
+    index = 1
+    for option in options:
+        key = option.get('text') + option.get('image_url') if option.get('image_url') else option.get('text')
+        if seen_options.get(key, 0) == 0:
+            missing_options.append('Option ' + str(index))
+        index += 1
+
+    if missing_options:
+        return {'seed_error': 'Missing option seed(s): ' + ', '.join(missing_options)}
+
+    return None
+
+
+def validate_seeded_answers_random(answers):
+    """
+    This validator checks if there is a minimum of one answer
+
+    Args:
+        answers (list): the answers to be checked
+
+    Returns:
+        None if everything is good. Otherwise, the missing option error message.
+    """
+    if len(answers) < 1:
+        return {'seed_error': 'Missing 1 option seed'}
+
+    return None
+
+
+def validate_seeded_answers(answers, options, algo):
+    """
+    Validate answers based on selection algorithm
+
+    This is called when instructor setup the tool and providing seeded answers to the question.
+    This function is trying to validate if instructor provided enough seeds for a give algorithm.
+    e.g. we require 1 seed for each option in simple algorithm and at least 1 seed for random
+    algorithm. Because otherwise, the first student won't be able to see the answers on the
+    second step where he/she suppose to compare and review other students answers.
+
+    Args:
+        answers (list): list of dict that contain seeded answers
+        options (dict): all options that should exist in the answers
+        algo (str): selection algorithm
+
+    Returns:
+        None if successful, otherwise error message
+    """
+    if algo['name'] == 'simple':
+        return validate_seeded_answers_simple(answers, options, algo)
+    elif algo['name'] == 'random':
+        return validate_seeded_answers_random(answers)
+    else:
+        raise UnknownChooseAnswerAlgorithm()
+
+
+def get_other_answers(pool, seeded_answers, get_student_item_dict, algo, options):
+    """
+    Select other student's answers from answer pool or seeded answers based on the selection algorithm
+
+    Args:
+        pool (dict): answer pool, format:
+            {
+                option1_index: {
+                    student_id: { can store algorithm specific info here }
```

Comment:
Do you mean in the `{ can store algorithm specific info here }`? It's not used for simple and random algorithm but it could be something like: {score: 5} or whatever an algorithm needs to calculate and make the selection decisions based on.